libnmstate/state.py

```diff
@@ -132,6 +188,20 @@ def verify_interfaces(self, other_state):
 
         self._assert_interfaces_equal(other_state)
 
+    def verify_routes(self, other_state):
+        self._clean_routes()
+        other_state._clean_routes()
+        for iface_name, routes in six.viewitems(self.iface_routes):
+            if routes != other_state.iface_routes.get(iface_name):
```

Comment:
Is the current requirement that the desired state needs to include all routes for a specific interface instead of only the routes that it wants to add (or remocve if state is absent)?

Reply:
You can just include routes you want without interface defined, but that interface should be exist/up in current state.

Routes will be ignored(warning message will show) for interface down/absent or IP disabled.

Reply:
With the interfaces, we first optimize to check that the desired interfaces are included in the other state (based on the names as the index and using the len() func). It does not assure it is indeed included, but it is a good first quick check. This probably can be applied here on the interface names as well (as the routes are indexed by them).

Then comes the original question from @tyll , which I could not understand the answer to.
Why are we not checking inclusion (like with the interfaces) but full equality of the routes records?